BEIMA.Backend/DeviceFunctions/AddDevice.cs

```diff
@@ -0,0 +1,73 @@
+using BEIMA.Backend.MongoService;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace BEIMA.Backend.DeviceFunctions
+{
+    public static class AddDevice
+    {
+        /// <summary>
+        /// Handles device create request.
+        /// </summary>
+        /// <param name="req">The http request.</param>
+        /// <param name="log">The logger to log to.</param>
+        /// <returns>An http response containing the id of the newly created device.</returns>
+        [FunctionName("AddDevice")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "device")] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation("C# HTTP trigger function processed a device post request.");
+
+            Device device;
+            try
+            {
+                string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+                dynamic data = JsonConvert.DeserializeObject(requestBody);
+                device = new Device(ObjectId.GenerateNewId(),
```

Comment:
We have to cast each property on the dynamic data variable, otherwise it can't seem to figure out what type they are supposed to be.